src/System.Runtime/tests/System/GuidTests.cs

```diff
@@ -35,12 +36,25 @@ public static void Ctor_ByteArray(byte[] b, Guid expected)
             Assert.Equal(expected, new Guid(b));
         }
 
+        [Theory]
+        [MemberData(nameof(Ctor_ByteArray_TestData))]
+        public static void Ctor_ReadOnlySpan(byte[] b, Guid expected)
+        {
+            Assert.Equal(expected, new Guid(new ReadOnlySpan<byte>(b)));
+        }
+
         [Fact]
         public static void Ctor_NullByteArray_ThrowsArgumentNullException()
         {
             AssertExtensions.Throws<ArgumentNullException>("b", () => new Guid((byte[])null));
         }
 
+        [Fact]
+        public static void Ctor_NullReadOnlySpan_ThrowsArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() => new Guid((ReadOnlySpan<byte>)null));
```

Comment:
This test should be removed as it's not actually testing any functionality of `Guid`. The new `Guid` constructor that takes a `ReadOnlySpan<byte>` doesn't throw `ArgumentNullException` -- there's no need because `ReadOnlySpan<byte>` is a struct that cannot be null. The code throwing `ArgumentNullException` is the `ReadOnlySpan<T>` constructor that takes a `T[]` when the array is null (called by the implicit conversion operator used here).